packages/testing-lib/src/waitForTransition/waitForTransition.ts

```diff
@@ -0,0 +1,21 @@
+import { fireEvent, waitFor } from '@testing-library/react';
+
+import { act } from '../RTLOverrides';
+
+/**
+ * Fires the `transitionEnd` event on the provided element,
+ * triggering any event handlers attached to this CSS transition events.
+ *
+ * Ensures `onEnter`,`onEntering`,`onEntered`, `onExit`, `onExiting` and `onExited` handlers are fired when
```

Comment:
From the comments in the test file above, `onEntered` and `onExited` are the only ones not called until `fireEvent.transitionEnd`. How does this ensure that the other handlers like `onEnter`, `onEntering`, etc, are fired?

Reply:
Ah, good catch. It only ensures the `*ed` handlers are fired. Updating the comment